### DISCUSSION TITLE
Is there any way to preserve existing url hashes on Router instantiation?

### DISCUSSION THREAD

**Discussion by geoff-harper** [TanStack/router#600] [Q&A] [answered]
I have an important hash in the url of the page where my react app loads. However that hash was being cleared immediately and it was causing issues with some other deps. After some trial and error I discovered that it was cleared when I called `new Router({...})`. Is there any way I can prevent that from being cleared? I don't expect the router to do anything with it, just not remove it. I took a peak through the router options and nothing jumped out at me for being a solution to my problem. If possible I'd like to stick with the `BrowserRouter`. Thanks in advance

**Comment by bhaugeea** [comment, marked as answer]
This seems to be a known bug. #300, #268, #561 (a PR), #542, #457.

**Reply by geoff-harper** [reply]
Apparently I've done a shockingly bad job looking for existing discussion. Thanks for the links
